### DISCUSSION TITLE
No network found?  Retry!

### DISCUSSION THREAD

**Discussion by Pontiac76** [tteck/Proxmox#2877] [Ideas]
I run a DHCP server that if a MAC address is asking for an IP address that I haven't statically assigned in my DHCP config, the IP that is given doesn't get a route to the internet.  This is a BY DESIGN function.

Given that, I ran the Homepage LCX script and ran into problems trying to get to connect to the internet for the obvious reason above.  This script is creating a random MAC address, brings up the interface, my DHCP server correctly feeds it an IP address with no route provided.

If there's other scripts, I'd like to offer the suggestion that wherever it's looking to see if there is a network connection (/Proxmox/main/misc/install.func - network_check()) that an option to RETRY the connection be available.  That'd mean restarting the network interface, or the DHCP client to ask for a new IP address with the same MAC.  It'd at least give an admin a chance to find out WHY the connection to the internet failed and retry instead of having to rebuild a new LCX.

In my case, I just switched to the advanced mode on the install and got around this, but the problem would probably persist if other installers run like this.

**Comment by tteck** [comment]
Your setup sounds unique. Not many do that type of setup.

**Reply by davidvermillion** [reply]
Not that unique. I ran into this exact issue and the scripts don't work because of it, and there doesn't appear to be a specify mac addresses to pre-populate an address either.

**Reply by Pontiac76** [reply]
I could get on board with having the option to specify a MAC (With or without the colons or dashes between the segments) either at the CLI before I hit ENTER on a paste, or, have the app give the option to specify a MAC in a network config of sorts.

**Comment by Pontiac76** [comment]
You betchya not many do.  The goal for my setup is because I don't trust drive by WiFi.  I want to know who's where, who's supposed to be around, etc.

But even then, if there's an issue with getting out to the internet, because I KNOW what my fault is going to be, someone else may run into another fault somewhere else and not be able to get out for one reason or another.

**Comment by halpcomputar** [comment]
At the beginning, if you select Advanced, you can select the MAC address that's provided. That's at least what I've found if you select a static IP configuration during setup.
<img width="577" alt="screenie" src="https://github.com/tteck/Proxmox/assets/74846313/5cae0807-f425-411d-8829-e9cb950e3895">




That being said, it sounds like you could really use a RADIUS server (and possibly some kind of EDR/MDR and proper VLAN management) in your life instead of painfully managing every MAC manually. Especially when it's trivial to sit outside, deauth wifi traffic, grab one of the MAC addresses I just deauthed and then just run macchanger to clone that MAC and still have someone nefarious in your network. I have a raspberry pi running in my truck right now 24/7 that can do that automatically with a couple quick scripts. You're putting in too much work for too little return with that setup.
